cmd/state/internal/cmdtree/shim.go

```diff
@@ -0,0 +1,30 @@
+package cmdtree
+
+import (
+	"strings"
+
+	"github.com/ActiveState/cli/internal/captain"
+	"github.com/ActiveState/cli/internal/locale"
+	"github.com/ActiveState/cli/internal/primer"
+	"github.com/ActiveState/cli/internal/runners/shim"
+)
+
+func newShimCommand(prime *primer.Values) *captain.Command {
+	runner := shim.New(prime)
+
+	cmd := captain.NewCommand(
+		"shim",
+		locale.T("shim_description"),
+		[]*captain.Flag{},
+		[]*captain.Argument{},
+		func(ccmd *captain.Command, args []string) error {
+			return runner.Run(args...)
+		},
+	)
+	cmd.SetSkipChecks(true)
+
+	if !strings.Contains(strings.Join(prime.Args(), " "), " -- ") {
+		cmd.SetDisableFlagParsing(true)
```

Comment:
This changed was prompted by running `state shim python -c ...`

If we leave flag parsing on Cobra will handle the `--` correctly, however since we have to support `state shim` without it we need to disable flag parsing when the delimiter isn't present so Cobra doesn't try to interpret flags for the shimmed command